components/Tracer/index.tsx

```diff
@@ -69,6 +74,7 @@ export const Tracer = ({ mainHeight }: TracerProps) => {
 
   const trace = tracerData?.trace
   const currentTraceEntry = tracerData?.trace[executionTraceStepNumber]
+  const errorTraceEntry = tracerData?.trace.at(-1)
```

Comment:
Set errorTraceEntry only if the executionState is an error.
```suggestion
  const errorTraceEntry = executionState === ProgramExecutionState.Error ? tracerData?.trace.at(-2) : null
```